cmd/root.go

```diff
@@ -163,9 +167,78 @@ func Config(cmd *cobra.Command) config.Config {
 	}
 	cfg.Path, _ = cmd.Flags().GetString("config")
 
+	// Validate report settings.
+	reportPath := getReportPath()
+	if reportPath != "" {
+		// TODO: improve this
+		//reportDir := filepath.Dir(reportPath)
+		//if d, err := os.OpenFile(reportDir, os.O_RDWR, 0); err != nil {
+		//	defer func() {
+		//		_ = d.Close()
+		//	}()
+		//	log.Fatal().Err(err).Str("report-path", reportPath).Msg("Invalid report path")
+		//}
+
+		var (
+			reporter       report.Reporter
+			reportFormat   = getReportFormat()
+			reportTemplate = getReportTemplate()
+		)
+		switch strings.ToLower(reportFormat) {
+		case "csv":
+			reporter = &report.CsvReporter{}
+		case "json":
+			reporter = &report.JsonReporter{}
+		case "junit":
+			reporter = &report.JunitReporter{}
+		case "sarif":
+			reporter = &report.SarifReporter{
+				OrderedRules: cfg.GetOrderedRules(),
+			}
+		case "template":
+			if reporter, err = report.NewTemplateReporter(reportTemplate); err != nil {
+				log.Fatal().Err(err).Msg("Invalid report template")
+			}
+		default:
+			log.Fatal().Msgf("unknown report format %s", reportFormat)
+		}
+
+		// Sanity check.
+		if reportTemplate != "" && reportFormat != "template" {
+			log.Fatal().Msgf("Report format must be 'template' if --report-template is specified")
+		}
+
+		cfg.ReportPath = reportPath
+		cfg.Reporter = reporter
+	}
+
 	return cfg
 }
 
+func getReportPath() string {
```

Comment:
I don't know if these are necessary. In what conditions can `rootCmd.Flags().GetX(...)` fail?